gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -145,6 +145,67 @@ public static BlobTargetOption metagenerationMatch() {
     public static BlobTargetOption metagenerationNotMatch() {
       return new BlobTargetOption(StorageRpc.Option.IF_METAGENERATION_NOT_MATCH);
     }
+
+    static BlobWriteOption[] convert(BlobTargetOption[] options, BlobWriteOption... optionsToAdd) {
+      BlobWriteOption[] writeOptions = new BlobWriteOption[options.length + optionsToAdd.length];
+      int index = 0;
+      for (BlobTargetOption option : options) {
+        writeOptions[index++] = new BlobWriteOption(option);
+      }
+      for (BlobWriteOption option : optionsToAdd) {
+        writeOptions[index++] = option;
+      }
+      return writeOptions;
+    }
+  }
+
+  class BlobWriteOption extends Option {
```

Comment:
See my comment above about trying to isolate this change in the service impl layer.

Did you consider the following:
1) Change `BlobWriteOptions` to implement `Serializable` rather than extend `Option`.
2) Have a package scope static method in `BlobWriteOption` that accepts `BlobInfo` and `BlobWriteOption...`
and returns a tuple (`BlobInfo`, `BlobTargetOpiton...`).
3) Add another private `create` method similar to the one that accepts `InputStream` but this one accepts `BlobTargetOption`.